src/DataStreams/RemoteQueryExecutor.cpp

```diff
@@ -180,7 +190,37 @@ Block RemoteQueryExecutor::read()
         if (context.getSettingsRef().skip_unavailable_shards && (0 == multiplexed_connections->size()))
             return {};
     }
+    try
+    {
+        return readPackets();
+    }
+    catch (const DB::Exception & ex)
+    {
+        if (ex.code() == ErrorCodes::DUPLICATED_PART_UUIDS)
```

Comment:
Not clear to me, but why we have to build this logic on top of exception handling? Why we cannot just return some flag or smth like this?

Reply:
Or this exception maybe also thrown for parts reader?